pysmt/factory.py

```diff
@@ -511,6 +511,7 @@ def get_implicant(self, formula, solver_name=None,
                         if solver.get_value(a).is_true():
                             res.append(a)
                         else:
+                            assert solver.get_value(a).is_false(), solver_name
```

Comment:
The assertion should probably raise the result of get_value, and not the solver name.